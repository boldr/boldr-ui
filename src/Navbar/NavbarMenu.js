/* @flow */
import * as React from 'react';
import classNames from 'classnames';
import { getActiveModifiers, removeActiveModifiers, createWrappedComponent } from '../util/boldrui';
import { StyleClasses } from '../theme/styleClasses';
import { getDomSafeProps } from '../util/helpers';

export type Props = {
  tag?: string,
  className?: string,
};

const BASE_ELEMENT = StyleClasses.NAVBAR_MENU;

export function NavbarMenu({ tag = 'div', ...props }: Props) {
  const className = classNames(
    BASE_ELEMENT,
    {
      ...getActiveModifiers(props),
    },
    props.className,
  );

  const HTMLProps = getDomSafeProps(props, removeActiveModifiers);

  return React.createElement(tag, { ...HTMLProps, className });
}

export default createWrappedComponent(NavbarMenu);
