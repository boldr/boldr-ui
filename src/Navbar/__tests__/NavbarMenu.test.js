import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import { NavbarMenu } from '../NavbarMenu';

describe('<NavbarMenu />', () => {
  it('should render a div with .boldr-navbar__menu', () => {
    const component = shallow(<NavbarMenu>My Nav</NavbarMenu>);
    expect(component.contains(<div className="boldr-navbar__menu">My Nav</div>)).toBe(true);
  });

  it('should render a div with .boldr-navbar__menu', () => {
    const component = shallow(<NavbarMenu tag="nav">My Nav</NavbarMenu>);
    expect(component.contains(<nav className="boldr-navbar__menu">My Nav</nav>)).toBe(true);
  });

  it('should render a div with .boldr-navbar__menu and modifiers', () => {
    const component = shallow(
      <NavbarMenu isActive>
        <span>Any Content</span>
      </NavbarMenu>,
    );
    expect(component.hasClass('boldr-navbar__menu')).toBe(true);
    expect(component.hasClass('is-active')).toBe(true);
  });

  it('should render a div with .boldr-navbar__menu, modifiers and custom classNames', () => {
    const component = shallow(
      <NavbarMenu isActive className="custom">
        <span>Any Content</span>
      </NavbarMenu>,
    );
    expect(component.hasClass('boldr-navbar__menu')).toBe(true);
    expect(component.hasClass('is-active')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });

  it('should render a div with .boldr-navbar__menu and custom classNames', () => {
    const component = shallow(
      <NavbarMenu className="custom">
        <span>Any Content</span>
      </NavbarMenu>,
    );
    expect(component.hasClass('boldr-navbar__menu')).toBe(true);
    expect(component.hasClass('custom')).toBe(true);
  });
  it('should match the snapshot', () => {
    const wrapper = shallow(<NavbarMenu>My Nav</NavbarMenu>);
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
