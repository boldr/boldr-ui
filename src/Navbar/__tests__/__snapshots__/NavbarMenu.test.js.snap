// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavbarMenu /> should match the snapshot 1`] = `
<div
  className="boldr-navbar__menu"
>
  My Nav
</div>
`;
